### PR TITLE
Add "is_temporary" field to narrative object metadata

### PR DESCRIPTION
**Pull request by briehl**

### CODE REVIEW

File: lib/NarrativeService/NarrativeManager.py [hunk 1]
```diff
@@ -207,6 +207,10 @@ def copy_narrative(self, newName, workspaceRef, workspaceId):
             newNarMetadata['ws_name'] = newWsName
             newNarMetadata['job_info'] = json.dumps({'queue_time': 0, 'running': 0,
                                                      'completed': 0, 'run_time': 0, 'error': 0})
+            is_temporary = 'false'
+            if newNarMetadata['name'] == 'Untitled' or newNarMetadata['name'] is None:
```
Comment (MrCreosote) [inline lib/NarrativeService/NarrativeManager.py]: Bill sez narratives cannot be called untitled when saving, so untitled is a magic title that has special semantics

File: lib/NarrativeService/NarrativeManager.py [hunk 2]
```diff
@@ -283,6 +287,11 @@ def _create_temp_narrative(self, cells, parameters, importData, includeIntroCell
         [narrativeObject, metadataExternal] = self._fetchNarrativeObjects(
             workspaceName, cells, parameters, includeIntroCell, title
         )
+        is_temporary = 'true'
```
Comment (MrCreosote) [inline lib/NarrativeService/NarrativeManager.py]: _create_temp_narrative, according to bill, is also used to create non-temporary narratives. That screams for a refactor but is out of scope here.

File: lib/NarrativeService/NarrativeManager.py (hunk 1 above)
Comment (MrCreosote) [inline lib/NarrativeService/NarrativeManager.py]: Should an existing is_temporary key override the title logic?

File: lib/NarrativeService/NarrativeManager.py (hunk 2 above)
Comment (briehl) [reply lib/NarrativeService/NarrativeManager.py]: This whole module screams for a refactor. Loudly. With klaxons and gongs and airhorns.

File: lib/NarrativeService/NarrativeManager.py (hunk 1 above)
Comment (briehl) [reply lib/NarrativeService/NarrativeManager.py]: Yes, it probably should. I'll put in a fix.

File: lib/NarrativeService/NarrativeManager.py [hunk 3]
```diff
@@ -207,6 +207,11 @@ def copy_narrative(self, newName, workspaceRef, workspaceId):
             newNarMetadata['ws_name'] = newWsName
             newNarMetadata['job_info'] = json.dumps({'queue_time': 0, 'running': 0,
                                                      'completed': 0, 'run_time': 0, 'error': 0})
+            if 'is_temporary' not in newNarMetadata:
```
Comment (MrCreosote) [inline lib/NarrativeService/NarrativeManager.py]: I don't think this is tested?

**Comment by briehl** [comment]
Made some changes to setting up the 'is_temporary' logic on narrative copy, added more tests.

**Comment by MrCreosote** [comment]
This looks ok. The case where a narrative is saved with an Untitled title but is not a temporary narrative (e.g. a user runs an app in an Untitled narrative) is not covered because per Bill that's a bug that should be fixed.

**Pull request merged**